### PR TITLE
Custom ActorSystem creation in MultiNodeSpec 

### PR DESCRIPTION
**Pull request by johanandren**
Refs #22180

Is this sufficient @jroper ?

### CODE REVIEW

File: akka-multi-node-testkit/src/main/scala/akka/remote/testkit/MultiNodeSpec.scala [hunk 1]
```diff
@@ -255,9 +255,16 @@ abstract class MultiNodeSpec(val myself: RoleName, _system: ActorSystem, _roles:
...
+   * the multi node spec (the `Config` passed to the creator must be used in
+   * the created actor system for the multi node tests to work)
+   */
+  def this(config: MultiNodeConfig, actorSystemCreator: Config ⇒ ActorSystem) =
+    this(config.myself, actorSystemCreator(ConfigFactory.load(config.config)), config.roles, config.deployments)
+
   def this(config: MultiNodeConfig) =
-    this(config.myself, ActorSystem(MultiNodeSpec.getCallerName(classOf[MultiNodeSpec]), ConfigFactory.load(config.config)),
-      config.roles, config.deployments)
+    this(config, config ⇒ ActorSystem(MultiNodeSpec.getCallerName(classOf[MultiNodeSpec]), config))
```
Comment (patriknw) [inline akka-multi-node-testkit/src/main/scala/akka/remote/testkit/MultiNodeSpec.scala]: Does `getCallerName` still work?

Comment (johanandren) [inline akka-multi-node-testkit/src/main/scala/akka/remote/testkit/MultiNodeSpec.scala]: Yep! However it is broken for all subclasses of `MultiNodeRemotingSpec` (by me as well, but a while ago). I created #22186 to fix that

Comment (patriknw) [inline akka-multi-node-testkit/src/main/scala/akka/remote/testkit/MultiNodeSpec.scala]: ok, good, yeah it's a horrible hack

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**